src/chart/generateCategoricalChart.tsx

```diff
@@ -736,8 +736,7 @@ const calculateOffset = (
   }
 
   if (legendItem && prevLegendBBox) {
-    // @ts-expect-error margin is optional in props but required in appendOffsetOfLegend
-    offset = appendOffsetOfLegend(offset, props, prevLegendBBox);
+    offset = appendOffsetOfLegend(offset, legendItem, prevLegendBBox);
```

Comment:
we only passed in props to get `legendItem` again using `findChildByType`, but we quite literally already have it above. Why find it twice? Don't!